core/embed/rust/src/ui/model_tt/bootloader/mod.rs

```diff
@@ -511,3 +531,158 @@ extern "C" fn screen_welcome() {
     show(&mut frame, false);
     fadein();
 }
+
+#[no_mangle]
+extern "C" fn screen_boot(
+    vimg: cty::uintptr_t,
+    vimg_datalen: u32,
+    vendor_str: *const cty::c_char,
+    vendor_str_len: u8,
+    version: *const cty::c_char,
+    vtrust: u16,
+) {
+    let img = if vimg_datalen != 0 {
+        unsafe {
+            Some(core::slice::from_raw_parts(
+                vimg as *const u8,
+                vimg_datalen as usize + TOIF_HEADER_LENGTH,
+            ))
+        }
+    } else {
+        None
+    };
+
+    let vendor_str = if vendor_str.is_null() {
+        None
+    } else {
+        unsafe { from_c_array(vendor_str, vendor_str_len as _) }
+    };
+
+    let version = if version.is_null() {
+        None
+    } else {
+        unsafe { from_c_str(version) }
+    };
+
+    fadeout();
```

Comment:
Any reason to do `fadeout()` here in the middle of assigning? I would prefer all the actions be together, so they are better visible - in this case above `display::rect_fill` probably

Reply:
https://github.com/trezor/trezor-firmware/commit/35391d98c6e13fa84c443ec624cd949e7964e1db